### PR TITLE
Allow bulk adding to all sessions

### PR DESCRIPTION
**Pull request by cameron1729**
**Prerequisites**: A course with some students enrolled

# Testing instructions
1. Create a face to face activity with default settings:
2. Press the "Save and display" button
3. Click the "Add a new session" link
4. Set "Session date/time known" to "Yes"
5. Set the start time and finish time to a date in the past
6. Press the "Save changes" button
7. Click the "Add a new session" link
8. Set "Session date/time known" to "Yes"
9. Set the start time and finish time to a date in the past
10. Press the "Add a new date" button
11. Set the start time and finish time of the new date to a date in the future
12. Press the "Save changes" button
13. Click the "Add a new session" link
14. Set "Session date/time known" to "Yes"
15. Set the start time and finish time to a date in the future
16. Press the "Save changes" button
17. Click the "Attendees" link for the session under the "Upcoming sessions" heading
18. Click the "Add/remove attendees" link
19. **Verify** there is **no** checkbox with the text "Add the selected users to all (upcoming) sessions"
20. Select a user and press the "Add" button
21. Click the "Go back" link
22. Click the "Go back" link
23. **Verify** that the capacity for the session under the "Upcoming sessions" header is now 1/10
24. **Verify** all other sessions show 0/10
25. Edit the face to face activity, changing the following settings:
    * "Signup type": "Multiple"
    * "Multiple signup method": "Per activity"
26. Press the "Save and display" button
27. Click the "Attendees" link for the session under "Upcoming sessions"
28. Click the "Add/remove attendees" link
29. **Verify** there **is** a checkbox with the text "Add the selected users to all (upcoming) sessions" (don't check it)
30. Select a user and press the "Add" button
31. Click the "Go back" link
32. Click the "Go back" link
33. **Verify** that the capacity for the session under the "Upcoming sessions" header is now 2/10
34. **Verify** all other sessions show 0/10
35. Click the "Attendees" link for the session under "Upcoming sessions"
36. Click the "Add/remove attendees" link
37. Check the "Add the selected users to all (upcoming) sessions" checkbox
38. Select a user and press the "Add" button
39. Click the "Go back" link
40. Click the "Go back" link
41. **Verify** that the capacity for the session under the "Upcoming sessions" header is now 3/10
42. **Verify** that the capacity for the session that has a date in the future is now 1/10 (even though this session is under the "Previous sessions" heading (this is possibly a pre-existing bug))
43. **Verify** the last remaining session still shows 0/10

### CODE REVIEW

**Comment by dmitriim** [comment]
Thanks @cameron1729 for working on that feature.

I tested and it works as intended. 
Few commends on UI/UX:
- is it possible to move a checkbox close to Add/Remove buttons? 
- the checkbox should be unchecked once you added users
- it says "Add the selected users to all (upcoming) sessions" which was confusing for me as you can select users in the Attendees part of the form. So maybe change the language to "Add users to all (upcoming) sessions". And remove selected.
- maybe update help text to be consistent to "Suppress email" one and make it more clear that this option will be used when adding users

File: signup.php [hunk 1]
```diff
@@ -128,7 +128,13 @@
     if ($isbulksignup) {
         $error = '';
         $message = get_string('bookingcompleted', 'facetoface');
-        foreach (facetoface_get_sessions($facetoface->id) as $session) {
+        $now = time();
```
Comment (dmitriim) [inline signup.php]: looks like the code below is a good candidate for a new function like "facetoface_get_future_session" to reduce the code duplication.

Comment (cameron1729) [reply signup.php]: Done, moved into a new function in lib.php

**Comment by cameron1729** [comment]
Thanks for review @dmitriim - I've added some commits addressing all of your points.

File: editattendees.php [hunk 2]
```diff
@@ -105,20 +106,30 @@
...
-                } else {
-                    $status = MDL_F2F_STATUS_WAITLISTED;
-                }
-                if (!facetoface_user_signup($session, $facetoface, $course, '', MDL_F2F_BOTH,
-                $status, $adduser, !$suppressemail)) {
-                    $erruser = $DB->get_record('user', array('id' => $adduser), "id, {$usernamefields}");
-                    $errors[] = get_string('error:addattendee', 'facetoface', fullname($erruser));
+                $now = time();
+                $addtofuturesessions = $addtoallsessions && $facetoface->signuptype == MOD_FACETOFACE_SIGNUP_MULTIPLE;
+                $futuresessions = array_filter(
```
Comment (dmitriim) [inline editattendees.php]: should we use facetoface_get_future_sessions here?

Comment (cameron1729) [reply editattendees.php]: Oops, yes we should. Fixed that.

**Comment by dmitriim** [comment]
I'll squash this one to make it easy for backporting...

**Pull request merged**